File: Python/7 kyu/Simple string division/main.py
def solve(st,k):
    longest = len(st) - k
    maxNum = 0
    
    for i in range(len(st) - longest + 1):
        num = int(st[i : i + longest])
        if num > maxNum:
            maxNum = num
    
    return maxNum

if __name__ == "__main__" :
    print(solve("1234", 2))

File: GoLang/7 kyu/Simple string division/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(Solve("1234", 2))
}

func Solve(st string, k int) int {
  lenOfNum := len(st) - k
  maxNum := 0
  
  for i := 0; i <= len(st) - lenOfNum; i++ {
    
    currentNumStr := st[i : i + lenOfNum]
    numInt, _ := strconv.Atoi(currentNumStr)
    
    if numInt > maxNum { 
      maxNum = numInt 
    }
    
  }
  
  return maxNum
}